cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/batch/source/DBSource.java

```diff
@@ -124,6 +131,48 @@ private String getJDBCPluginId() {
     return String.format("%s.%s.%s", "source", dbSourceConfig.jdbcPluginType, dbSourceConfig.jdbcPluginName);
   }
 
+  private void ensureValidConnection(Class<? extends Driver> jdbcDriverClass) {
+    try {
+      ensureJDBCDriverIsAvailable(jdbcDriverClass);
+    } catch (Exception e) {
```

Comment:
Again, better to propagate the exception, so you don't have so many try..catch blocks. You can just have the call to ``ensureValidConnection`` in a try...catch.